core/Piwik.php

```diff
@@ -889,4 +889,22 @@ public static function translate($translationId, $args = array(), $language = nu
 
         return $translator->translate($translationId, $args, $language);
     }
+
+    public static function getPeriod($paramName = 'period', $default = false)
+    {
+        $period = Common::getRequestVar($paramName, $default);
+        if (is_array($period)) {
+            $period = implode(',', $period);
+        }
```

Comment:
There is no need fo that special handling. If it's and array, it's fine if the `Common::getRequestVar` above throws and exception. As it is simply unsupported and was manually provided by the user. Concatenating would cause other issues later, which would need to handle again.